tests/TestPsci/EvalTest.hs

```diff
@@ -0,0 +1,67 @@
+module TestPsci.EvalTest where
+
+import Prelude ()
+import Prelude.Compat
+
+import           Control.Monad (forM_, foldM_)
+import           Control.Monad.IO.Class (liftIO)
+import           Data.List (isPrefixOf, intercalate)
+import           Data.List.Split (splitOn)
+import           System.Directory (getCurrentDirectory)
+import           System.Exit (exitFailure)
+import           System.FilePath ((</>), takeFileName)
+import qualified System.FilePath.Glob as Glob
+import           System.IO.UTF8 (readUTF8File)
+import           Test.HUnit
+import           TestPsci.TestEnv
+
+evalTests :: Test
+evalTests = TestLabel "evalTests" . TestCase $ do
+  putStrLn "\nPSCi eval test suites"
+  testFiles <- evalTestFiles
+  forM_ testFiles evalTest
+
+evalTestFiles :: IO [FilePath]
+evalTestFiles = do
+  cwd <- getCurrentDirectory
+  let psciExamples = cwd </> "examples" </> "psci"
+  Glob.globDir1 (Glob.compile "**/*.purs") psciExamples
+
+data EvalLine = Line String
+              | Comment EvalContext
+              | Empty
+              | Invalid String
+              deriving (Show)
+
+data EvalContext = ShouldEvaluateTo String
+                 | Paste [String]
+                 | None
+                 deriving (Show)
+
+evalCommentPrefix :: String
+evalCommentPrefix = "-- @"
+
+parseEvalLine :: String -> EvalLine
+parseEvalLine "" = Empty
+parseEvalLine line
+  | evalCommentPrefix `isPrefixOf` line =
+    case splitOn " " $ drop (length evalCommentPrefix) line of
+      "shouldEvaluateTo" : args -> Comment (ShouldEvaluateTo $ intercalate " " args)
+      "paste" : [] -> Comment (Paste [])
+      _ -> Invalid line
+  | otherwise = Line line
+
+evalTest :: FilePath -> IO ()
+evalTest f = do
+  putStrLn $ "  " ++ takeFileName f
```

Comment:
It was a title of each test case. Not we don't need this because Hspec's `specify` is used instead.